hydrolib/profile_optimizer/profile_optimizer/postprocessing.py

```diff
@@ -0,0 +1,141 @@
+    def export_result_gpkg(self, output_dir):
+        self.gdf_lines.to_file(str(Path(output_dir)/'Results_on_lines.gpkg'), driver='GPKG')
+        self.gdf_points.to_file(str(Path(output_dir) / 'Results_on_points.gpkg'), driver='GPKG')
+
+    def export_result_shp(self, output_dir):
+        self.gdf_lines.to_file(str(Path(output_dir)/'Results_on_lines.shp'))
+        self.gdf_points.to_file(str(Path(output_dir) / 'Results_on_points.shp'))
+
+def plot_profiles(bron_model, talud_profiel, geoptimaliseerde_bodembreedte, waterdiepte, profile_id):
+    #HUIDIG model
+    base_model = FMModel(bron_model)
+    cross_def = pd.DataFrame([cs.__dict__ for cs in base_model.geometry.crossdeffile.definition])
+    profielen_yz = cross_def[cross_def['type'] == 'yz']
+    profielen_yz = profielen_yz[profielen_yz['id'] == profile_id]
+    
+    #normaliseren aan de hand van laagste z punt
+    profielen_yz['zcoordinates_normalized'] = ''
+    profielen_yz['ycoordinates_normalized'] = ''
+
+    for i in range(len(profielen_yz['id'])):
+        #normalizeren over laagste punt z
+        profielen_yz['zcoordinates_normalized'].iloc[i] = profielen_yz['zcoordinates'].iloc[i] - (np.ones(len(profielen_yz['zcoordinates'].iloc[i])) * min(profielen_yz.zcoordinates.iloc[i]))
+        
+        #normalizeren over y bij dat laagste punt z
+        index_laagste_z = profielen_yz.zcoordinates.iloc[i].index(min(profielen_yz.zcoordinates.iloc[i]))
+        profielen_yz['ycoordinates_normalized'].iloc[i] = profielen_yz['ycoordinates'].iloc[i] - np.ones(len(profielen_yz['ycoordinates'].iloc[i])) * profielen_yz['ycoordinates'].iloc[i][index_laagste_z]
+
+    #GEOPTIMALISEERD model
+    geoptimaliseerd_z = [waterdiepte, 0, 0, waterdiepte]
+    geoptimaliseerd_y = [geoptimaliseerde_bodembreedte * -0.5 - talud_profiel * waterdiepte,
+                         geoptimaliseerde_bodembreedte * -0.5,
+                         geoptimaliseerde_bodembreedte * 0.5,
+                         geoptimaliseerde_bodembreedte * 0.5 + talud_profiel * waterdiepte]
+
+    #FIGURE PROFIELEN
+    fig = go.Figure()
+    #huidige model
+    fig.add_trace(go.Scatter(x=profielen_yz.ycoordinates_normalized.iloc[0], y=profielen_yz.zcoordinates_normalized.iloc[0], mode='lines+markers', name='huidig profiel'))
+
+    #geoptimaliseerde model
+    fig.add_trace(go.Scatter(x=geoptimaliseerd_y, y=geoptimaliseerd_z, mode='lines+markers', name='geoptimaliseerd profiel'))
+    
+    #figure layout
+    fig.update_layout(title=profielen_yz.id.iloc[0])
+    fig.update_xaxes(title_text='y', title_font=dict(size=18, color='crimson'))
+    fig.update_yaxes(title_text='z', title_font=dict(size=18, color='crimson'))
+    
+    return fig
+
+if __name__ == '__main__':
+    crossdef_filename_huidig_model = 'playground/temp_new/Demo_8/crossdef_8.ini'
+    talud_profiel = 2
+    geoptimaliseerde_b = 4.269748446347192
+
+    folder = r'c:\Dev\profileoptimizer\playground\model\DFM_OUTPUT_moergestels_broek.mdu_Profile_Optimizer'
```

Comment:
The input for a lot of the scripts depend on local paths etc. Maybe it is worth to consider a configuration file for this tool including the input parameters? Or use an arg parser so that you can call it from the command line.

Reply:
This is left as future work for now, to proceed with the merge.